### PR TITLE
Edit user profile

### PR DESCRIPTION
**Pull request by MichaelParadis**
Bennett and Kieter were fine with the UI all that is need is code review. I will take care of the merge conflicts if needed.

### CODE REVIEW

**Comment by Hreherch** [comment]
Issue: The text box appears when not in edit mode.
How to:
1. press edit button
2. edit some text
3. save or cancel
4. press the text box again, and the keyboard will reappear. (can't edit it)

This is no bueno. 🌮

**Comment by MichaelParadis** [comment]
@Hreherch I have no idea why that happens. I am unable to bring up the keyboard unless I am in the edit mode. Although I have encountered a  bug with the email edit text which allows the changing of the text while not in edit mode. This is done by:
1. edit email
2. cancel or save without closing the keyboard
3. Start typing with the keyboard still up.

Although I can replicate this bug there is no reason that I can find for this happening since it only happens with the email edit text. I am unable to replicate this when I try to edit the phone number text.

**Comment by Hreherch** [comment]
I have fixed the merge conflict with master.

I has also fixed the issues I had with the UI. Note that I had to remove the highlighting of the EditText, but I made a cursor workaround that should be okay for now. 

I will say that I believe the code can be cleaned up and generalized, as there is a lot of functions that do very similar things. At the very least, there should be at most three (edit, save, delete) functions and at best two (edit, save+delete). Right now there are six different functions and editing one means editing at least one other one... this is a low priority issue. @kieter and @MichaelParadis did a good job getting this initialized. 

Furthermore, please immediately fix a merge conflict when you make a pull request in the future. 

**I approve this branch to be merged with master.** Someone else can do it, they don't need to check it over but this allows them to.

File: app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java [hunk 1]
```diff
@@ -1,38 +1,73 @@
...
 import android.view.View;
+import android.view.inputmethod.InputMethodManager;
 import android.widget.EditText;
 import android.widget.ImageButton;
 import android.widget.TextView;
 
-/**
- * Created by kiete on 10/18/2016.
- */
+// TODO A lot can be done to reduce code duplication. Notice how there are 4 functions that close,
```
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java:16]: I concur

**Comment by kieter** [comment]
@MichaelParadis please review the changes Bennett made in this pull request, they're good 👍 .

File: app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java [hunk 2]
```diff
@@ -108,10 +154,14 @@ public void editEmailAddress(View v){
         cancelButton.setVisibility(View.VISIBLE);
         EditText emailView = (EditText) findViewById(R.id.EmailEditText);
         // Set it so the user can edit the EditText
-        emailView.setFocusable(true);
+        emailView.setFocusableInTouchMode(true);
```
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java:157]: I just looked this up. It's interesting, what "Focus" and "FocusInTouchMode" is.

File: app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java [hunk 3]
```diff
@@ -1,38 +1,73 @@
...
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_user_profile);
+        UserController uc = new UserController();
+        User currentUser = UserController.getLoggedInUser();
+
+        String username;
+
+        // Developer code, may not be needed for release...
+        if (currentUser == null) {
```
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java:39]: This is pretty handy.

**Pull request merged**